src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java

```diff
@@ -125,14 +128,23 @@ public ClickHouseNode getServer() {
     public void doInsert(List<Record> records) {
         if ( records.isEmpty() )
             return;
-        Record first = records.get(0);
-        switch (first.getSchemaType()) {
-            case SCHEMA:
-                doInsertRawBinary(records);
-                break;
-            case SCHEMA_LESS:
-                doInsertJson(records);
-                break;
+
+        try {
+            Record first = records.get(0);
+            switch (first.getSchemaType()) {
+                case SCHEMA:
+                    doInsertRawBinary(records);
+                    break;
+                case SCHEMA_LESS:
+                    doInsertJson(records);
+                    break;
+            }
+        } catch (DataException | NullPointerException dataException) {
+            LOGGER.debug("Exception in doInsert", dataException);
+            throw dataException;
+        } catch (Exception e) {
+            LOGGER.debug("Exception in doInsert", e);
+            throw new RetriableException(e);
```

Comment:
The documentation doesn't provide many details. https://kafka.apache.org/23/javadoc/org/apache/kafka/connect/errors/RetriableException.html 
What exceptions are considered to be retriable?